### PR TITLE
build(odsp-doclib-utils): convert to fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
semi-automated

### CODE REVIEW

**Comment by sonalideshpandemsft** [comment]
can `isomorphic-fetch` be used instead of `node-fetch`? `node-fetch` is not compatible with both env together

**Comment by jason-ha** [comment]
> If I bump to the latest version of 'node-fetch', CJS builds fail with:

Any chance odsp-doclib-utils could be changed to ESM only? (just move forward)

File: packages/utils/odsp-doclib-utils/src/test/tsconfig.cjs.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	// This config must be used in a "type": "commonjs" environment. (Use tsc-mult.)
```
Comment (jason-ha) [inline packages/utils/odsp-doclib-utils/src/test/tsconfig.cjs.json:2]: Same general comment update requested x2. tsc-multi -> fluid-tsc commonjs

**Comment by sonalideshpandemsft** [comment]
> > If I bump to the latest version of 'node-fetch', CJS builds fail with:
> 
> Any chance odsp-doclib-utils could be changed to ESM only? (just move forward)

We might have to use the same dependency for any other package using node-fetch. In those cases too, isomorphic-fetch can be an alternative.

**Review by sonalideshpandemsft — APPROVED** [review]

File: packages/utils/odsp-doclib-utils/package.json [hunk 2]
```diff
@@ -11,31 +11,32 @@
...
 		".": {
 			"import": {
-				"types": "./lib/odsp-doclib-utils-public.d.mts",
-				"default": "./lib/index.mjs"
+				"types": "./lib/index.d.ts",
+				"default": "./lib/index.js"
 			},
 			"require": {
-				"types": "./dist/odsp-doclib-utils-public.d.ts",
+				"types": "./dist/index.d.ts",
```
Comment (jason-ha) [inline packages/utils/odsp-doclib-utils/package.json:22]: If we already have public in place, don't roll it back to "internal".

File: packages/utils/odsp-doclib-utils/package.json [hunk 3]
```diff
@@ -114,6 +118,7 @@
 		"@fluidframework/driver-utils": "workspace:~",
 		"@fluidframework/odsp-driver-definitions": "workspace:~",
 		"@fluidframework/telemetry-utils": "workspace:~",
+		"isomorphic-fetch": "^3.0.0",
 		"node-fetch": "^2.6.9"
 	},
```
Comment (jason-ha) [inline packages/utils/odsp-doclib-utils/package.json:123]: keep both?

File: packages/utils/odsp-doclib-utils/src/odspRequest.ts [hunk 4]
```diff
@@ -3,8 +3,9 @@
  * Licensed under the MIT License.
  */
 
-import { fetch } from "./fetch";
-import { IOdspAuthRequestInfo, authRequestWithRetry } from "./odspAuth";
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (jason-ha) [inline packages/utils/odsp-doclib-utils/src/odspRequest.ts:6]: lint disables should have a comment explaining

File: packages/utils/odsp-doclib-utils/src/odspRequest.ts [hunk 5]
```diff
@@ -13,7 +14,10 @@ export async function getAsync(
 	url: string,
 	authRequestInfo: IOdspAuthRequestInfo,
 ): Promise<Response> {
-	return authRequest(authRequestInfo, async (config: RequestInit) => fetch(url, config));
+	return authRequest(
+		authRequestInfo,
+		async (config: RequestInit) => fetch(url, config) as unknown as Response,
```
Comment (jason-ha) [inline packages/utils/odsp-doclib-utils/src/odspRequest.ts:19]: ouch - what is going on here?

**Review by jason-ha — COMMENTED** [review]
I am not feeling the love for isomorphic-fetch.

File: packages/utils/odsp-doclib-utils/package.json (hunk 3 above)
Comment (DLehenbauer) [reply packages/utils/odsp-doclib-utils/package.json:112]: Nope.  Just need one.  (Fixed).

File: packages/utils/odsp-doclib-utils/package.json (hunk 2 above)
Comment (DLehenbauer) [reply packages/utils/odsp-doclib-utils/package.json]: Oops.  Thanks.

File: packages/utils/odsp-doclib-utils/src/odspRequest.ts (hunk 4 above)
Comment (DLehenbauer) [reply packages/utils/odsp-doclib-utils/src/odspRequest.ts]: Don't think we need the disable.  (Removed)

File: packages/utils/odsp-doclib-utils/src/odspRequest.ts (hunk 5 above)
Comment (DLehenbauer) [reply packages/utils/odsp-doclib-utils/src/odspRequest.ts]: Simplified to just 'as Response' (ditto elsewhere).

**Pull request merged**